test/test_respondent_controller.py

```diff
@@ -1215,6 +1215,26 @@ def test_put_email_verification_uses_case_insensitive_email_query(self):
             account_controller.put_email_verification(token)
             query.assert_called_once_with("test@example.test", db.session())
 
+    def test_token_removed_on_email_update(self):
+        respondent = self.populate_with_respondent(respondent=self.mock_respondent_with_pending_email)
+        token = self.generate_valid_token_from_email(respondent.pending_email_address)
+        self.put_email_verification(token, 200)
+        respondent = respondents()[0]
+        self.assertIsNone(respondent.password_verification_token)
+        self.assertEqual(respondent.password_reset_counter, 0)
+
+    def test_email_verification_no_password_reset_token(self):
```

Comment:
Do we need this function? does it not have line coverage already?

Reply:
It's more to check if the section doesn't fall over when there is no verification token as the rest of the tests have a token by default

Reply:
Ok, it's not really needed as all you are doing is removing the token and then running the code and making sure it's not there. Not major, but I would removed